pkg/apis/submarinerconfig/v1alpha1/types.go

```diff
@@ -46,12 +46,46 @@ type SubmarinerConfigSpec struct {
 	// preparation, this field should be specified.
 	// +optional
 	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// SubscriptionConfig represents a Submariner subscription. SubscriptionConfig
+	// can be used to customize the Submariner subscription.
+	// +optional
+	SubscriptionConfig `json:"subscriptionConfig,omitempty"`
+}
+
+// SubscriptionConfig contains configuration specified for a submariner subscription.
+type SubscriptionConfig struct {
+	// CatalogSource represents the source of a submariner subscription.
+	// The default value is redhat-operators
+	// +optional
+	CatalogSource string `json:"source,omitempty"`
+
+	// CatalogSourceNamespace represents the sourceNamespace of a submariner subscription.
+	// The default value is openshift-marketplace
+	// +optional
+	CatalogSourceNamespace string `json:"sourceNamespace,omitempty"`
+
+	// Name represents the name of a submariner subscription.
+	// The default value is submariner
+	// +optional
+	Name string `json:"name,omitempty"`
```

Comment:
will we have a different  name?

Reply:
only one case, if I create a CatalogSource by myself, I can change the name...